imageio/plugins/ffmpeg.py

```diff
@@ -471,24 +514,27 @@ def _open(self, fps=10, codec='libx264', bitrate=400000):
             # Initialize parameters
             self._proc = None
             self._size = None
-        
-        def _close(self):
-            # Close subprocess
-            if self._proc is not None:
-                self._proc.stdin.close()
-                self._proc.stderr.close()
-                self._proc.wait()
-                self._proc = None
-        
+            self._cmd = None
+
+        def _close(self, timeout=1.0):
+            if self._proc is None:  # pragma: no cover
+                return  # no process
+            if self._proc.poll() is not None:
+                return  # process already dead
+            # End process
+            # self._proc.stdin.close() # .communicate will take care of this
+            # The .communicate is preferred over .wait which can deadlock
+            self._proc.communicate()
```

Comment:
I guess in looking at what the above .communicate source does, we can essentially go back to stdin.close() and .wait(). Looks no different. Maybe it's clearer that way.